src/java/org/apache/cassandra/net/OutboundTcpConnection.java

```diff
@@ -505,6 +508,10 @@ private boolean connect()
             {
                 logger.error("SSL handshake error for outbound connection to " + socket, e);
                 socket = null;
+                if (DatabaseDescriptor.isSslNteEnabled()) {
```

Comment:
Want to confirm - this will only catch SSL exceptions that originate on _this_ node right? Is there any chance that these exceptions might cause other nodes/the whole cluster to shut down too?

Reply:
Shit, yeah you're right... this could throw if another server's certs are bad, and would cause a cluster wide outage :(

Reply:
I'll need to rethink this approach

Reply:
maybe this is a use case for a _transient error_ where we don't kill the server on failure? our only goal is to surface and prevent the cluster from rolling (change the to "not ready")